fsspec/implementations/zip.py

```diff
@@ -132,3 +132,59 @@ def _open(
             out.size = info["size"]
             out.name = info["name"]
         return out
+
+    def find(self, path, maxdepth=None, withdirs=False, detail=False, **kwargs):
+        if maxdepth is not None and maxdepth < 1:
+            raise ValueError("maxdepth must be at least 1")
+
+        result = {}
+
+        def _below_max_recursion_depth(path):
+            if not maxdepth:
+                return True
+
+            depth = len(path.split("/"))
+            return depth <= maxdepth
+
+        for zip_info in self.zip.infolist():
+            file_name = zip_info.filename
+            if not file_name.startswith(path.lstrip("/")):
```

Comment:
```suggestion
            if not file_name.startswith(path.lstrip("/")) + "/":
```
? 

So that find('apath') doesn't match 'apathlonger', only 'apath/nested'.

Reply:
Yeah. That makes sense.